src/plugins/intel_gna/src/transformations/insert_copy_layer.cpp

```diff
@@ -115,6 +116,43 @@ InsertCopyBeforeConcatLayer::InsertCopyBeforeConcatLayer() {
     this->register_matcher(m, callback);
 }
 
+InsertCopyBeforeBroadcastLayer::InsertCopyBeforeBroadcastLayer() {
+    MATCHER_SCOPE(InsertCopyBeforeBroadcastLayer);
+
+    auto broadcast_op = ngraph::pattern::wrap_type<ngraph::opset9::Broadcast>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto broadcast = m.get_match_root();
+
+        std::set<std::shared_ptr<ngraph::Node>> inputs;
+        // Insert copy layers after concat inputs with multiple connections to concat
+        for (size_t i = 0; i < broadcast->get_input_size(); i++) {
+            auto input_op = broadcast->get_input_node_shared_ptr(i);
+
+            if (inputs.find(input_op) != inputs.end()) {
+                insert_copy_layer_between(input_op, broadcast, i);
+            } else {
+                inputs.insert(input_op);
+            }
+        }
+
+        for (size_t i = 0; i < broadcast->get_input_size(); i++) {
```

Comment:
If we are looking for Broadcast node connected with Input maybe it is easier to use the following template:
    const auto input = ngraph::pattern::wrap_type<Parameter>(ngraph::pattern::type_matches_any(kSupportedInputTypesFrom));
    const auto broadcast_op = ngraph::pattern::wrap_type<Broadcast>({input});

Reply:
agreed to not change due the fact we need also handle non-functional layers between.